latch/registry/project.py

```diff
@@ -0,0 +1,55 @@
+import latch.gql as gql
+from latch import Account
+
+
+class Project:
+    def __init__(self, project_id: str):
+        self.id = project_id
+
+        self._name = None
+
+    @property
+    def name(self):
+        if self._name is not None:
+            return self._name
+
+        self._name = gql.execute(
+            document="""
+                query projectNameQuery ($argProjectId: BigInt!) {
+                    catalogProject(id: $argProjectId) {
+                        displayName
+                    }
+                }
+            """,
+            variables={"argProjectId": self.id},
+        )["catalogProject"]["displayName"]
+
+        return self._name
+
+    @classmethod
+    def from_name(cls, account: Account, project_name: str):
```

Comment:
`from_id` seems unnecessary since thats literally a restatement of the constructor, and since we're nixing the `from_name` in favor of `Account.list_projects`, theres nothing to be symmetric with